tests/lib/test_Repo.py

```diff
@@ -10,12 +10,12 @@
 #
 # instantiation
 #
-variants = {
-    'str': 'the-repo',
-    'Path': Path('the-repo'),
-}
-@pytest.mark.parametrize('variant', variants.values(), ids=variants.keys())
-def test_Repo_instantiate_types(tmp_path, variant):
+@pytest.mark.parametrize('variant', ['the-repo', Path('the-repo')])
+def test_Repo_instantiate_types(tmp_path: str, variant: Union[Path, str]) -> None:
+    """
+    Test that the Repo class instantiates correctly for different data types
```

Comment:
Nit: the phrase "Test that" will be repeated for every docstring. This is unnecessary. We're in the tests. We can accept that it's testing functionality.

For example, rather than "Test that the range is higher than 10." simply state: "The range must be higher than 10."

The tests in many ways are a code version of the specification. Imagine that the tests/docstrings are written based off the spec. Let them be declarative.